### PR TITLE
Add msgGasLimit support

### PR DESCRIPTION
**Pull request by nmlinaric**
## Description
- adds `msgGasLimit` support for new `permissionlessGenericHandler` msg format
- updates e2e test with new Solidity contracts changes

## Related Issue Or Context
Closes: #10 

## How Has This Been Tested? Testing details.
 - unit tests
- e2e tests
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

**Review by MakMuftic — APPROVED** [review]

File: api/api.go [hunk 1]
```diff
@@ -138,12 +146,13 @@ func (h *Handler) getRate(c *gin.Context) {
...
 		FromDomainID:             fromDomainID,
 		ToDomainID:               toDomainID,
+		MsgGasLimit:              msgGasLimit,
 		DataTimestamp:            dataTime,
 		SignatureTimestamp:       signedTime,
 		ExpirationTimestamp:      h.dataExpirationManager(dataTime),
 	}
 
-	endpointRespData.Signature, err = h.rateSignature(endpointRespData, fromDomainID, resource.ID)
+	endpointRespData.Signature, err = h.rateSignature(endpointRespData, fromDomainID, "0x0000000000000000000000000000000000000000000000000000000000000000")
```
Comment (mpetrun5) [inline api/api.go:155]: Why was this changed?

Comment (mpetrun5) [reply api/api.go:155]: It does seem wrong

File: api/api.go [hunk 2]
```diff
@@ -129,6 +130,13 @@ func (h *Handler) getRate(c *gin.Context) {
 		}
 	}
 
+	msgGasLimitParam := c.DefaultQuery("msgGasLimit", "0")
+	msgGasLimit, err := util.MsgGasLimitChecker(msgGasLimitParam)
```
Comment (mpetrun5) [inline api/api.go:134]: Not really sure what this function does by the name.
If it only converts from string to *big.Int then it doesn't really have anything to do with gasLimit and it is only FormatBigInt()

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: api/api.go (hunk 2 above)
Comment (nmlinaric) [reply api/api.go:134]: It converts string to *big.Int and then check if msgGasLimit is less than 0

File: api/api.go [hunk 3]
```diff
@@ -129,6 +130,13 @@ func (h *Handler) getRate(c *gin.Context) {
 		}
 	}
 
+	msgGasLimitParam := c.DefaultQuery("msgGasLimit", "0")
+	checkIntegerRegex := regexp.MustCompile(`^[0-9]+$`)
```
Comment (MakMuftic) [inline api/api.go:134]: We can extract this outside function so it is not compiled on every API call.

Comment (MakMuftic) [inline api/api.go:134]: We can extract this outside function so it is not compiled on every API call.

**Review by MakMuftic — CHANGES_REQUESTED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Review by MakMuftic — APPROVED** [review]
